src/realm.h

```diff
@@ -3178,8 +3178,9 @@ RLM_API bool realm_app_push_notification_client_deregister_device(const realm_ap
  * @return true, if no error occurred.
  */
 RLM_API bool realm_app_call_function(const realm_app_t*, const realm_user_t*, const char* function_name,
-                                     const char* serialized_ejson_args, realm_return_string_func_t callback,
-                                     realm_userdata_t userdata, realm_free_userdata_func_t userdata_free);
+                                     const char* serialized_ejson_args, const char* service_name,
```

Comment:
Looks good, but you need to fix format checks and add a comment as per @fealebenpae request. 
`service_name` can be `NULL`.

Reply:
The [check for null is here](https://github.com/realm/realm-core/blob/a137e617a2e21a963e27486b849f4ff90b76a0a1/src/realm/object-store/c_api/app.cpp#L625).
` util::Optional<std::string> service_name_opt = service_name ? util::some<std::string>(service_name) : util::none;`